core/trino-main/src/main/java/io/trino/operator/scalar/TDigestFunctions.java

```diff
@@ -59,8 +59,14 @@ public static Block valuesAtQuantiles(@SqlType(StandardTypes.TDIGEST) TDigest in
         return output.build();
     }
 
+    public static void verifyValue(double value)
+    {
+        checkCondition(Double.isFinite(value), INVALID_FUNCTION_ARGUMENT, "value must be finite; was %s", value);
```

Comment:
This will cause `value` to be boxed to a Double on every invocation. In the past, we've seen this cause performance problems in some cases.

Reply:
Hmm - did not think of that. 

Would having 
```
    public static void checkCondition(boolean condition, ErrorCodeSupplier errorCode, Supplier<String> errorMessage) {
        if (!condition) {
            throw new TrinoException(errorCode, errorMessage.get());
        }
    }
```

and then doing 
```
checkCondition(Double.isFinite(value), INVALID_FUNCTION_ARGUMENT, () -> String.format("value must be finite; was %s", value));
```

help?

Would boxing happen only on call to lambda then. I think so but not 100% sure.

Reply:
I added extra commit